### PR TITLE
[SPARK-48490][CORE][FOLLOWUP] Properly process escape sequences

### PR DESCRIPTION
**Pull request by gengliangwang**
### What changes were proposed in this pull request?

Even with the fix in https://github.com/apache/spark/pull/46824, the escape sequences (`\r`, `\n`, `\t`, etc) are not handled properly. For example, when we use `log"\n"`, the StringContext interprets `\n` as a literal backslash `\` followed by `n` instead of a newline character. As a result, the bytes of `log"\n".message` becomes `[92, 110]`, instead of `[10]`.

This PR is to fix the issue by using the method StringContext.processEscapes in `LogStringContext`.

### Why are the changes needed?

 To ensure that escape sequences are properly processed in Spark logs

### Does this PR introduce _any_ user-facing change?

No
### How was this patch tested?

New UT

### Was this patch authored or co-authored using generative AI tooling?

No

### CODE REVIEW

**Comment by gengliangwang** [comment]
@panbingkun

**Comment by panbingkun** [comment]
+1, LGTM.

**Comment by panbingkun** [comment]
Wait, there may still be some minor issues. Let me verify them in UT.

**Review by HyukjinKwon — APPROVED** [review]
LGTM pending tests

File: common/utils/src/main/scala/org/apache/spark/internal/Logging.scala
Comment (panbingkun) [inline common/utils/src/main/scala/org/apache/spark/internal/Logging.scala:146]: I think this following also needs to be dealt with:
image

Comment (panbingkun) [reply common/utils/src/main/scala/org/apache/spark/internal/Logging.scala:146]: log"This is a log message\nThis is a new line \t other msg ${MDC(LogKeys.PATHS, "C:\\Users\\run-all_1.R\nC:\\Users\\run-all_2.R")} This is a 2 log message\nThis is a 2 new line \t other 2 msg"

Comment (panbingkun) [reply common/utils/src/main/scala/org/apache/spark/internal/Logging.scala:146]: Before:
```

```


After:
```
24/06/21 11:10:02 ERROR PatternLoggingSuite: This is a log message
This is a new line 	 other msg C:\Users\run-all_1.R
C:\Users\run-all_2.R This is a 2 log message
This is a 2 new line 	 other 2 msg
```

**Review by panbingkun — CHANGES_REQUESTED** [review]
```
 sb.append(processedParts.next()) 
```
->
```
 sb.append(StringContext.processEscapes(processedParts.next()))
```

File: common/utils/src/main/scala/org/apache/spark/internal/Logging.scala
Comment (gengliangwang) [reply common/utils/src/main/scala/org/apache/spark/internal/Logging.scala:146]: Thanks, updated

Comment (panbingkun) [reply common/utils/src/main/scala/org/apache/spark/internal/Logging.scala:146]: @gengliangwang
I have submitted the verified UT as a separate PR here.
https://github.com/apache/spark/pull/47051

Comment (panbingkun) [reply common/utils/src/main/scala/org/apache/spark/internal/Logging.scala:146]: Currently, it looks very perfect.

**Review by panbingkun — APPROVED** [review]

**Comment by panbingkun** [comment]
@HyukjinKwon Can we merge this?
After this PR is merged, I will test whether `R Windows` has been fixed.

**Pull request closed without merging**
